sys/include/net/ng_netif/hdr.h

```diff
@@ -134,6 +136,46 @@ static inline void ng_netif_hdr_set_dst_addr(ng_netif_hdr_t *hdr, uint8_t *addr,
     memcpy(((uint8_t *)(hdr + 1)) + hdr->src_l2addr_len, addr, addr_len);
 }
 
+/**
+ * @brief   Builds a generic network interface header for sending and
+ *          adds it to the packet buffer.
+ *
+ * @author  Martine Lenders <mlenders@inf.fu-berlin.de>
```

Comment:
Yes. The rest of this header is by @haukepetersen.
And yes, of course it's possible.